app/ide-desktop/lib/client/src/preload.js

```diff
@@ -46,3 +47,27 @@ if (win !== undefined) {
 
 // TODO[WD] Enable after making preload configurable (we do not want to load it always)
 // require('devtron').install()
+
+// =====================
+// === Lifecycle API ===
+// =====================
+
+contextBridge.exposeInMainWorld('enso_lifecycle', {
+    quit: data => ipcRenderer.send('quit-ide'),
```

Comment:
what `enso_lifecycle` is? I cannot find any docs in the codebase about it

Reply:
The new `--save-profile` option is not just for batch mode; it can also be used in interactive mode, to get a profile file instead of dumping the data into the console. In order to streamline this usage, I've introduced a keybinding that captures a profile and exits: Ctrl+Alt+Q. (I will be updating developer docs [soon](https://www.pivotaltracker.com/story/show/182048373) to include the new, much easier, profiling processes).

The new IPC APIs are all documented at the binding definitions, in `debug_api`: https://github.com/enso-org/enso/blob/24dd07d12fe2ab4d10fe136300e9f63aaf5c470e/lib/rust/debug-api/src/lib.rs#L22-L48

Reply:
Cool! Is this documented somewhere in the code as well? I'm thinking about the use case that someone (developer) is reading this file, sees this line and tries to understand why it is used for. I think this line should have at least short docs pointing to longer ones (placed where the definition of this mechanism lives).

Reply:
Added docs giving brief explanations of the new APIs, and referring to the `debug_api` module for primary documentation.